src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -517,6 +557,36 @@ func (h *PromWriteHandler) parseRequest(
 		}
 	}
 
+	if h.remoteWriteOpts.rejectOldSamples {
+		now := h.nowFn()
+		thresholdTime := now.Add(-h.remoteWriteOpts.rejectDuration)
+		// The two for loops do the trick of not allocating new memory and do in place drops
+		i := 0
+		for _, ts := range req.Timeseries {
+			j := 0
+			var t int64
+			for _, s := range ts.Samples {
+				t = s.Timestamp
+				if time.UnixMilli(s.Timestamp).After(thresholdTime) {
```

Comment:
sample timestamp is millisecond according to prometheus protocol: https://prometheus.io/docs/concepts/remote_write_spec/#protocol